src/test/java/seedu/address/logic/commands/FindConcertContactCommandTest.java

```diff
@@ -0,0 +1,167 @@
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_PERSON;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.function.Predicate;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.concert.Concert;
+import seedu.address.model.concert.ConcertContact;
+import seedu.address.model.person.Person;
+
+/**
+ * Contains integration tests (interaction with the Model) for
+ * {@code FindConcertContactCommand}.
+ */
+public class FindConcertContactCommandTest {
+
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+    private Model expectedModel = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void equals() {
+        Index personIndex = INDEX_FIRST_PERSON;
+        Index personIndex2 = INDEX_SECOND_PERSON;
+        Index concertIndex = INDEX_FIRST_CONCERT;
+
+        FindConcertContactCommand findFirstCommand = new FindConcertContactCommand(personIndex, concertIndex);
+        FindConcertContactCommand findSecondCommand = new FindConcertContactCommand(personIndex, concertIndex);
+        FindConcertContactCommand findThirdCommand = new FindConcertContactCommand(personIndex2, concertIndex);
+
+        // same object -> returns true
+        assertTrue(findFirstCommand.equals(findFirstCommand));
+
+        // same values -> returns true
+        assertTrue(findFirstCommand.equals(findSecondCommand));
+
+        // different types -> returns false
+        assertFalse(findFirstCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(findFirstCommand.equals(null));
+
+        // different index -> returns false
+        assertFalse(findFirstCommand.equals(findThirdCommand));
+    }
+
+    @Test
+    public void execute_indexOutOfBounds_throwsCommandException() {
+        // Person index too big
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredPersonList().size() + 1);
+        FindConcertContactCommand findCommand = new FindConcertContactCommand(outOfBoundIndex, null);
+        assertCommandFailure(findCommand, model, MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+
+        // Concert index too big
+        outOfBoundIndex = Index.fromOneBased(model.getFilteredConcertList().size() + 1);
+        findCommand = new FindConcertContactCommand(null, outOfBoundIndex);
+        assertCommandFailure(findCommand, model, MESSAGE_INVALID_CONCERT_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_personKeyword_noConcertContactsFound() {
+        String expectedMessage = String.format(MESSAGE_CONCERT_CONCERTS_LISTED_OVERVIEW, 0);
+        FindConcertContactCommand command = new FindConcertContactCommand(Index.fromZeroBased(2), null);
+        Person p = getTypicalAddressBook().getPersonList().get(2);
```

Comment:
Maybe for all these literals should take from TypicalIndexes instead, cos the other command tests also do like that
i.e. INDEX_SECOND_PERSON.getZeroBased()

Reply:
Done